Convert docs to docusaurus and build sidebar

elliot-nelson opened the pull request:
### SUMMARY

This PR is the first "interesting" one:

 - Fix the frontmatter in doc files to match docusaurus
 - Build a sidebar for the Docs pages
 - Fix up base docusaurus configuration
 - Convert the "Who's involved" page from Jekyll-y HTML to a React-based MDX page
 - Remove most of the docusaurus starter files

You should review and merge #3 before reviewing this PR.

octogonz commented on websites/rushstack.io/components/GitHubCard.jsx:
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+export function GitHubCard(props) {
```
We should use Heft+ESLint+TypeScript for this project.  It doesn't need to happen in this PR, but we should address it soon.  The community has been asking for real world examples of React projects using our stack.

octogonz commented on websites/rushstack.io/docs/heft/core_plugins.md:
```diff
@@ -10,20 +8,20 @@ These plugins are bundled with the `@rushstack/heft` package and are enabled by
 | [ProjectValidatorPlugin](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/ProjectValidatorPlugin.ts) | An internal Heft plugin that performs basic validation such as warning about obsolete files in the `.heft` folder |
-| [SassTypingsPlugin](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/SassTypingsPlugin/SassTypingsPlugin.ts) | Implements the [sass-typings]({% link pages/heft_tasks/sass-typings.md %}) task |
-| [TypeScriptPlugin](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/TypeScriptPlugin/TypeScriptPlugin.ts) | Implements the [typescript]({% link pages/heft_tasks/typescript.md %}), [eslint]({% link pages/heft_tasks/eslint.md %}), and [tslint]({% link pages/heft_tasks/tslint.md %}) tasks |
+| [SassTypingsPlugin](https://github.com/microsoft/rushstack/blob/master/apps/heft/src/plugins/SassTypingsPlugin/SassTypingsPlugin.ts) | Implements the [sass-typings](../heft_tasks/sass-typings) task |
```
This file is using `/heft_tasks/` and `../heft_tasks/` inconsistently.

🤔 Since Docusaurus is going to (presumably) normalize the output URls, I wonder if we should standardize on `/` rooted paths everywhere.  That way if you need to bulk search+replace `(/heft_tasks/api-extractor` you could have confidence that you fixed them all.  

Although I suppose if Docusaurus reports broken links, maybe this problem isn't too difficult.

elliot-nelson replied:
In order to convert all these {% link %} objects I used a home-grown "converter" tool that I haven't checked in yet (it's sitting in repo-scripts in my local copy), and it probably just needs a little more work.

Docusaurus recommends you _always use relative paths_ for doc-to-doc links.  This is because as you enable additional nested layers of docs (first you start versioning, then you start translating, etc.), they _do not_ fake new root doc URLs for you, so using only relative paths is the best way to ensure each "section" of the docs is correctly linked up to itself.

(So, that said, I just need to fix up the non-relative-links.)

octogonz commented on websites/rushstack.io/docs/heft/architecture.md:
```diff
@@ -1,15 +1,13 @@
 - **action** - a Heft command-line verb (as formalized by Rush Stack's [ts-command-line](https://www.npmjs.com/package/@rushstack/ts-command-line) system).  For example, the "test" action appears in the  `heft test --clean` command.  Actions are the user interface, essentially macros tailored for typical developer activities.
 - **plugin** - a TypeScript class that implements the `IHeftPlugin` contract to provide arbitrary functionality.  Plugins are loaded optionally based on Heft's configuration.
-- **plugin package** - an NPM package that provides one or more Heft plugins.  You can create your own NPM package that exports custom plugins for Heft to use.  See [Core Plugins]({% link pages/heft/core_plugins.md %}) for a list of the built-in plugins.
+- **plugin package** - an NPM package that provides one or more Heft plugins.  You can create your own NPM package that exports custom plugins for Heft to use.  See [Core Plugins](core_plugins) for a list of the built-in plugins.
```
After years of staring at TypeScript imports, it feels slightly weird to see `[Core Plugins](core_plugins)` instead of `[Core Plugins](./core_plugins)`.  I guess it's normal for relative website URIs. 

(I checked CommonMark and they don't say anything beyond specifying that this transforms into an `<a href...` tag.)

elliot-nelson replied:
I think we can choose to enforce `./` if we want, it should work either way.

octogonz replied:
I like the idea of enforcing that links whose target is under `pages` should start with `./` or `../`.  I wonder if Docusaurus has a built-in linter that would make this cheap to implement.

octogonz commented on websites/rushstack.io/docusaurus.config.js:
```diff
@@ -95,23 +128,30 @@ const config = {
       },
+      algolia: {
+        appId: 'W2G1E3U5T0',
+        apiKey: 'a0ab6dfc3db0c301b0ca8e725af85641',
```
Does this work? Might be interesting to test it.

elliot-nelson replied:
It's the settings from our current app, but I suspect testing it will be hard until the site is on the approved domain.

(An admin for our algolia account might be able to log in and preemptively crawl one of the deployed test pages though, I'm not sure what options you have there.)

octogonz replied:
Lemme investigate

octogonz commented on websites/rushstack.io/people.json:
```diff
@@ -0,0 +1,99 @@
+{
```
Should this file go in a subfolder such as `data`?  Generally we avoid putting nonstandard files in a project's root folder.

octogonz commented on websites/rushstack.io/components/People.jsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
What is the difference between `components` and `src`?

octogonz commented:
This PR is really awesome!  Thanks for doing all this work! 😁

octogonz commented on websites/rushstack.io/docusaurus.config.js:
```diff
@@ -1,30 +1,37 @@
 // @ts-check
```
What does `@ts-check` do?

elliot-nelson commented:
Additional change made before merging:

 - Moved `/components` into `/src/components`.
 - Moved `people.json` and `api_nav.json` into `/data`.
 - Ran `rushx start` locally and confirmed site still functions.
